src/esp/assets/ResourceManager.cpp

```diff
@@ -180,31 +188,85 @@ bool ResourceManager::loadStage(
+
+      RenderAssetInstanceCreation creation{
+          .filepath = semanticStageFilename,
+          .scale = Cr::Containers::NullOpt,
+          .isStatic =
+              stageAttributes->getFrustrumCulling(),  // only treat as static if
+                                                      // doing culling
+          .isRGBD = false,
+          .isSemantic = true,
+          .lightSetupKey = NO_LIGHT_KEY};
+
+      bool semanticStageSuccess =
+          loadStageInternal(semanticInfo,  // AssetInfo
+                            &creation,
+                            &semanticRootNode,    // parent scene node
+                            &semanticDrawables);  // drawable group
+
+      // regardless of load failure, original code still changed
+      // activeSemanticSceneID_
+      if (!semanticStageSuccess) {
+        LOG(ERROR) << " ResourceManager::loadStage : Semantic Stage mesh "
+                      "load failed.";
+        return false;
+      } else {
+        LOG(INFO) << "ResourceManager::loadStage : Semantic Stage mesh : "
+                  << semanticStageFilename << " loaded.";
+      }
+    } else {  // semantic file name does not exist but house does
+      LOG(WARNING)
+          << "ResourceManager::loadStage : Not loading semantic mesh - "
+             "File Name : "
+          << semanticStageFilename << " does not exist.";
+    }
+  } else {  // not wanting to create semantic mesh
+    LOG(INFO) << "ResourceManager::loadStage : Not loading semantic mesh";
+  }
+  // save active semantic scene ID so that simulator can consume
+  activeSceneIDs[1] = activeSemanticSceneID;
+  const bool isSeparateSemanticScene = activeSceneIDs[1] != activeSceneIDs[0];
+
   auto& sceneGraph = sceneManagerPtr->getSceneGraph(activeSceneIDs[0]);
   auto& rootNode = sceneGraph.getRootNode();
   auto& drawables = sceneGraph.getDrawables();
 
-  // bool computeAbsoluteAABBs =
-  //   (info.type == AssetType::FRL_PTEX_MESH ||
-  //    info.type == AssetType::MP3D_MESH || info.type == AssetType::UNKNOWN ||
-  //    (info.type == AssetType::INSTANCE_MESH && splitSemanticMesh));
-
   AssetInfo renderInfo = assetInfoMap.at("render");
-  // pass nullptr as physics manager for render mesh, since we are loading
-  // collision mesh next
-  bool renderMeshSuccess =
-      loadStageInternal(renderInfo,         // AssetInfo
-                        &rootNode,          // parent scene node
-                        &drawables,         //  drawable group
-                        true,               // compute Absolute AABBs or not
-                        false,              // split semantic mesh or not
-                        renderLightSetup);  // light setup
 
+  RenderAssetInstanceCreation renderCreation{
+      .filepath = renderInfo.filepath,
+      .scale = Cr::Containers::NullOpt,
+      .isStatic = true,
+      .isRGBD = true,
+      .isSemantic = !isSeparateSemanticScene,
+      .lightSetupKey = renderLightSetupKey};
```

Comment:
Apologies if I'm not up-to-date on C++ standard requirements in Habitat, but designated initializers are standard only since C++20 or require a particular GNU extension, which may hurt portability (especially if we'd want to reconsider Windows support again). I'd personally stay with standard C++ with no compiler-specific extensions.

Same in other places (I spotted one or two more occurence at least).